From 91cc054d85812aad1a3203f7a009b1b2083df074 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Fri, 10 May 2024 13:14:43 +0800
Subject: [PATCH 69/89] mtk: mac80211: do not check pre-CAC allowed for scan

When scanning, interfaces only leave the channels in a very short time,
which can be tolerated. Therefore we do not check pre-CAC allowed for
scan.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 net/mac80211/scan.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/mac80211/scan.c b/net/mac80211/scan.c
index 0b8d69f..c6dcffa 100644
--- a/net/mac80211/scan.c
+++ b/net/mac80211/scan.c
@@ -582,8 +582,10 @@ static bool __ieee80211_can_leave_ch(struct ieee80211_sub_if_data *sdata)
 	if (!ieee80211_is_radar_required(local))
 		return true;
 
+	/* FIXME do not check pre-CAC allowed for scan.
 	if (!regulatory_pre_cac_allowed(local->hw.wiphy))
 		return false;
+	*/
 
 	list_for_each_entry(sdata_iter, &local->interfaces, list) {
 		for_each_valid_link(&sdata_iter->wdev, link_id)
-- 
2.18.0

